### PR TITLE
Fixes SI-6189.

### PR DESCRIPTION
**Pull request by axel22**
Disable @static for the REPL code.
The problem was that there are no companion classes generated for objects
that contain the top-level declarations in the REPL.
When there is no companion class, the selecting a @static field will translate
to a getter call, instead of to a field load.

Review by @paulp

### CODE REVIEW

File: src/compiler/scala/tools/nsc/backend/icode/GenICode.scala [hunk 1]
```diff
@@ -880,12 +880,16 @@ abstract class GenICode extends SubComponent  {
 
         case app @ Apply(fun @ Select(qual, _), args)
         if !ctx.method.symbol.isStaticConstructor 
-        && fun.symbol.isAccessor && fun.symbol.accessed.hasStaticAnnotation =>
+        && fun.symbol.isAccessor && fun.symbol.accessed.hasStaticAnnotation
+        && qual.tpe.typeSymbol.orElse(fun.symbol.owner).companionClass != NoSymbol =>
```
Comment (paulp) [inline src/compiler/scala/tools/nsc/backend/icode/GenICode.scala:884]: qual.tpe.typeSymbol.orElse(fun.symbol.owner) looks a bit sketchy, but I don't have anything concrete to propose.

**Comment by paulp** [comment]
Does this commit fix SI-6236 as well? Either way, this LGTM for now.  I have some work on @static which uncovered some other bugs but I need to organize it.

**Comment by axel22** [comment]
```
package p {
  object x {
    object x {
      @scala.annotation.static val foo: Int = 353
    }
  }
}
```

No, it does not solve SI-6236 - for some strange reason I'm trying to figure out, in separate compilation, although the `static` field gets generated, it's not loaded into the list of members symbol for `p.x$x`.

**Comment by axel22** [comment]
Ok, I've added a fix for SI-6236.

Maybe we should merge this into 2.10.x, given that all tests pass.

**Comment by paulp** [comment]
I have a suspicion we aren't done with @static.  I cherry-picked these commits into 2.10.x.

**Comment by axel22** [comment]
Ok, thanks!
If you observe any other problems with `@static`, please let me know!

**Pull request closed without merging**